packages/amplify-provider-awscloudformation/lib/attach-backend.js

```diff
@@ -0,0 +1,299 @@
+  if (!backendEnv) {
+    process.exit(1);
+  }
+
+  await downloadBackend(context, backendEnv, awsConfig);
+  const currentAmplifyMeta = await ensureAmplifyMeta(context, amplifyApp, awsConfig);
+
+  context.exeInfo.projectConfig.projectName = amplifyApp.name;
+  context.exeInfo.localEnvInfo.envName = backendEnv.environmentName;
+  context.exeInfo.teamProviderInfo[backendEnv.environmentName] = currentAmplifyMeta.providers;
+}
+
+async function ensureAmplifyMeta(context, amplifyApp, awsConfig) {
+  // check if appId is present in the provider section of the metadata
+  // if not, it's a migration case and we need to
+  // 1. insert the appId
+  // 2. upload the metadata file and the backend config file into the deployment bucket
+  const currentAmplifyMetaFilePath = context.amplify.pathManager.getCurrentAmplifyMetaFilePath(process.cwd());
+  const currentAmplifyMeta = context.amplify.readJsonFile(currentAmplifyMetaFilePath);
+  if (!currentAmplifyMeta.providers[constants.ProviderName][constants.AmplifyAppIdLabel]) {
+    currentAmplifyMeta.providers[constants.ProviderName][constants.AmplifyAppIdLabel] = amplifyApp.appId;
+
+    const amplifyMetaFilePath = context.amplify.pathManager.getAmplifyMetaFilePath(process.cwd());
+    const jsonString = JSON.stringify(currentAmplifyMeta, null, 4);
+    fs.writeFileSync(currentAmplifyMetaFilePath, jsonString, 'utf8');
+    fs.writeFileSync(amplifyMetaFilePath, jsonString, 'utf8');
+
+    const { DeploymentBucketName } = currentAmplifyMeta.providers[constants.ProviderName];
+    await storeArtifactsForAmplifyService(context, awsConfig, DeploymentBucketName);
+  }
+
+  return currentAmplifyMeta;
+}
+
+async function storeArtifactsForAmplifyService(context, awsConfig, deploymentBucketName) {
+  const s3Client = getConfiguredS3Client(awsConfig);
+  const amplifyMetaFilePath = context.amplify.pathManager.getCurrentAmplifyMetaFilePath(process.cwd());
+  const backendConfigFilePath = context.amplify.pathManager.getCurrentBackendConfigFilePath(process.cwd());
+  await uploadFile(s3Client, deploymentBucketName, amplifyMetaFilePath);
+  await uploadFile(s3Client, deploymentBucketName, backendConfigFilePath);
+}
+
+async function uploadFile(s3Client, bucketName, filePath) {
+  if (!fs.existsSync(filePath)) {
+    return;
+  }
+  const key = path.basename(filePath);
+  const body = fs.createReadStream(filePath);
+  await s3Client
+    .putObject({
+      Bucket: bucketName,
+      Key: key,
+      Body: body,
+    })
+    .promise();
+}
+
+async function getAmplifyApp(context, amplifyClient) {
+  // If appId is in the inputParams, verify it
+  const { inputParams } = context.exeInfo;
+  if (inputParams.amplify && inputParams.amplify.appId) {
+    const inputAppId = inputParams.amplify.appId;
+    try {
+      const getAppResult = await amplifyClient
+        .getApp({
+          appId: inputAppId,
+        })
+        .promise();
+      context.print.info(`Input Amplify appId is verified: ${inputAppId}`);
```

Comment:
This code block is also duplicated. We we extract it to a common place?

Reply:
not sure where you refer to as duplicate, will chat.

Reply:
Amplify AppID found: ${inputAmplifyAppId}. Amplify App name is ${getAppResult.app.name}`

Reply:
changed